### ISSUE TITLE
prefer-flat-map should not trigger on .flat() with a param >1

### ISSUE THREAD

**Issue by Sembiance** [freaktechnik/eslint-plugin-array-func#61]
Array.flat() has an optional 'depth' parameter (default of 1).
Array.flatMap() only supports the default depth of 1.

So if you have code:
```
myArray.flat(99)
```
The rule shouldn't trigger on that, but it currently does.

So basically if the .flat() call has an argument and that argument is greater than 1, then don't report it as an error as .flatMap() won't work in this situation.
